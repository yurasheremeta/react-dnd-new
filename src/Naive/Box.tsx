import * as React from 'react'
import { DragSource, ConnectDragSource } from 'react-dnd'
import ItemTypes from './ItemTypes'

const style: React.CSSProperties = {
	position: 'absolute',
	// border: '1px dashed gray',
	backgroundColor: 'white',
	// padding: '0.5rem 1rem',
	padding: '3px',
	cursor: 'move',

}

const boxSource = {
	beginDrag(props: BoxProps) {
		const { id, left, top, value } = props;
		return { id, left, top, value }
	},
}

export interface BoxProps {
	id: any
	left: number
	top: number
	hideSourceOnDrag?: boolean
	value: string

}

interface BoxCollectedProps {
	connectDragSource: ConnectDragSource
	isDragging?: boolean
}

class Box extends React.Component<BoxProps & BoxCollectedProps> {
	public render() {
		const {
			hideSourceOnDrag,
			left,
			top,
			connectDragSource,
			isDragging,
			children,
			id

		} = this.props
		if (isDragging && hideSourceOnDrag) {
			return null
		}

		const cStyle = { ...style, left, top };

		if (!left && !top) {
			cStyle.position = 'relative';
		}
		return connectDragSource(
			<div style={{ ...style, left, top }} id={id} className="boxDiv">
				{children}
			</div>
		)
	}
}

export default DragSource<BoxProps, BoxCollectedProps>(
	ItemTypes.BOX,
	boxSource,
	(connect, monitor) => ({
		connectDragSource: connect.dragSource(),
		isDragging: monitor.isDragging(),
	}),
)(Box)
