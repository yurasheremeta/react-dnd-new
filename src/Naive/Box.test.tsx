import * as React from 'react';
import { shallow, mount } from 'enzyme';
import Box from './Box';
import 'jsdom-global/register';

describe('Box tests' , () => {
    const props = {
        id: "a",
        left: 0,
        top: 40,
        value: "Hello world"
    }

    it('Find Boxes' , () => {
        const wrapper = shallow(<Box {...props}/>)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('div')).toBeDefined();
    })

    it('Render box props' , () => {
        const wrapper = mount(<Box {...props}/>);
        expect(wrapper.prop("id")).toEqual("a");
        expect(wrapper.prop("left")).toEqual(0);
        expect(wrapper.prop("top")).toEqual(40);
        expect(wrapper.prop("value")).toEqual("Hello world");
    })
  
    it('alow us to set props', () => {
        const wrapper = mount(<Box {...props}/>);
        expect(wrapper.prop("id")).toEqual("a");
        expect(wrapper.prop("left")).toEqual(0);
        expect(wrapper.prop("top")).toEqual(40);
        expect(wrapper.prop("value")).toEqual("Hello world");
        wrapper.setProps({ 
            id: "b",
            left: 20,
            top: 60,
            value: "Hell Yeah!"
        });
        expect(wrapper.prop("id")).toEqual("b");
        expect(wrapper.prop("left")).toEqual(20);
        expect(wrapper.prop("top")).toEqual(60);
        expect(wrapper.prop("value")).toEqual("Hell Yeah!");
    })
    
    // it('should display items, by default, in a text template (span element)', function() {
    //     var container = mount(<Box {...props}/>);
    //     expect(container.getDOMNode().textContent).toBe(randomWords.join(''));
    //   });
})