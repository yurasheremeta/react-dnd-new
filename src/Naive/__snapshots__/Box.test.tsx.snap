// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box tests Find Boxes 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Box tests render correctly Box component 1`] = `null`;
